src/util/psbt/serialize.rs

```diff
@@ -194,20 +205,15 @@ impl Serialize for schnorr::SchnorrSig  {
 
 impl Deserialize for schnorr::SchnorrSig {
     fn deserialize(bytes: &[u8]) -> Result<Self, encode::Error> {
-        match bytes.len() {
-            65 => {
-                let hash_ty = SchnorrSigHashType::from_u8(bytes[64])
-                    .map_err(|_| encode::Error::ParseFailed("Invalid Sighash type"))?;
-                let sig = secp256k1::schnorr::Signature::from_slice(&bytes[..64])
-                    .map_err(|_| encode::Error::ParseFailed("Invalid Schnorr signature"))?;
-                Ok(schnorr::SchnorrSig{ sig, hash_ty })
-            }
-            64 => {
-                let sig = secp256k1::schnorr::Signature::from_slice(&bytes[..64])
-                    .map_err(|_| encode::Error::ParseFailed("Invalid Schnorr signature"))?;
-                    Ok(schnorr::SchnorrSig{ sig, hash_ty: SchnorrSigHashType::Default })
+        match schnorr::SchnorrSig::from_slice(&bytes) {
+            Ok(sig) => Ok(sig),
+            Err(schnorr::SchnorrSigError::InvalidSighashType(flag)) => {
+                Err(encode::Error::from(psbt::Error::NonStandardSigHashType(flag as u32)))
             }
-            _ => Err(encode::Error::ParseFailed("Invalid Schnorr signature len"))
+            Err(schnorr::SchnorrSigError::InvalidSchnorrSigSize(_)) =>
+                Err(encode::Error::ParseFailed("Invalid Schnorr signature length")),
+            Err(schnorr::SchnorrSigError::Secp256k1(..)) =>
+                Err(encode::Error::ParseFailed("Invalid Schnorr signature")),
```

Comment:
This could've been `map_err` which would communicate the intent a bit better. Only realized it now.

Reply:
Issue: https://github.com/rust-bitcoin/rust-bitcoin/issues/793